### PR TITLE
[440006] added tests for rich string validation and compilation 

### PR DESCRIPTION
**Pull request by cdietrich**
[440006] added tests for rich string validation and compilation 

xtend tests for https://github.com/eclipse/xtext-extras/pull/31

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=440006

File: org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerBug440006Test.xtend [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+			        _builder.append(i, "");
+			      }
+			    }
+			    return _builder;
+			  }
+			}
+		''')
+	}
+	
+	def private for1() {
```
Comment (svenefftinge) [inline org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerBug440006Test.xtend]: I don't understand why the for loops are extracted to methods?

Comment (cdietrich) [inline org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerBug440006Test.xtend]: i assume this me me not knowing a better way to escape the <<>>

Comment (cdietrich) [inline org.eclipse.xtend.core.tests/src/org/eclipse/xtend/core/tests/compiler/CompilerBug440006Test.xtend]: i  just tested it, `«"'''«FOR i:#{1..2}»«i»«ENDFOR»'''"»` works as well

**Comment by cdietrich** [comment]
updated the pull request

**Pull request merged**